tests/integration/test_upgrade.py

```diff
@@ -0,0 +1,260 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Integration tests for upgrading WordPress charm."""
+
+import asyncio
+import copy
+import io
+import json
+import logging
+import re
+import subprocess  # nosec
+import time
+
+import kubernetes
+import ops.model
+import PIL.Image
+import pytest
+import pytest_operator.plugin
+import requests
+from playwright.async_api import async_playwright
+
+from tests.integration.wordpress_client_for_test import WordpressClient
+
+logger = logging.getLogger()
+
+POST_CONTENT = """\
+WordPress powers more than 39% of the web — a figure that rises every day.
+Everything from simple websites, to blogs, to complex portals and enterprise
+websites, and even applications, are built with WordPress. WordPress combines
+simplicity for users and publishers with under-the-hood complexity for
+developers. This makes it flexible while still being easy-to-use."""
+
+POST_CONTENT = POST_CONTENT.replace("\n", " ")
+
+POST_TITLE = "WordPress Post #1"
+
+POST_COMMENT = "I am a comment."
+
+
+async def screenshot(url, path):
+    """Create a screenshot of a website."""
+    async with async_playwright() as playwright:
+        browser = await playwright.chromium.launch()
```

Comment:
Would it be worth it to re-use a running chromium between tests? E.g., make this a fixture that cleans up chromium after the tests?

Reply:
It will complicate the process, e.g. managing tabs opened by tests. It's not that slow to open a new browser.